### PR TITLE
Only add xdebug if we dont have it set already on appveyor

### LINKED ISSUE #377: Config_Framework E2E test fails on AppVeyor
| Question    | Answer
| ------------| ---------------
| Infection version | master
| Test Framework version | PhpSpec
| PHP version | any
| Platform    | Windows + AppVeyor

Config_Framework E2E test sporadically fails on AppVeyor with the following explanation:

```
1) Infection\Tests\Console\E2ETest::test_it_runs_an_e2e_test_with_success with data set "Config_Framework" (Symfony\Component\Finder\SplFileInfo Object (...))
Unexpected exit code. Command output wasYou are running Infection with xdebug enabled.
 [WARNING] Infection cannot control exit codes and unable to relaunch itself.                                           
           It is your responsibility to disable xdebug/phpdbg unless needed.                                            
    ____      ____          __  _
   /  _/___  / __/__  _____/ /_(_)___  ____ 
   / // __ \/ /_/ _ \/ ___/ __/ / __ \/ __ \
 _/ // / / / __/  __/ /__/ /_/ / /_/ / / / /
/___/_/ /_/_/  \___/\___/\__/_/\____/_/ /_/
 
Running initial test suite...
PhpSpec version: 4.3.0
    0 [>---------------------------] < 1 sec
    1 [->--------------------------] < 1 sec
    2 [-->-------------------------] < 1 sec [ERROR] Project tests must be in a passing state before running Infection.                                             
                                                                                                                        
         PhpSpec reported an exit code of 1.                                                                            
                                                                                                                        
         Refer to the PhpSpec's output below:                                                                           
                                                                                                                        
         STDOUT:                                                                                                        
                                                                                                                        
         TAP version 13                                                                                                 
                                                                                                                        
                                                                                                                        
         STDERR:                                                                                                        
                                                                                                                        
         Cannot load Xdebug - it was already loaded

```

### PR DESCRIPTION
**Pull request by BackEndTea**
This should fix the appveyor failures

Fixes #377

### CODE REVIEW

**Comment by borNfreee** [comment]
Could you please explain what is going on on appveyor?

are you trying to fix [this](https://ci.appveyor.com/project/borNfreee/infection/build/1.0.767/job/8ojb37x8xdflox4o#L233) issue?

**Comment by borNfreee** [comment]
More interesting, master is green ;) https://ci.appveyor.com/project/borNfreee/infection/branch/master

**Comment by BackEndTea** [comment]
> are you trying to fix this issue?

That was the goal, yeah. 

I'm gonna keep this open in case master fails again due to the same error.

**Comment by sanmai** [comment]
[It is failing now.](https://ci.appveyor.com/project/borNfreee/infection/branch/master)

         PhpSpec reported an exit code of 1. 
                                               
         Refer to the PhpSpec's output below:     

         STDOUT:    

         TAP version 13 

    
         STDERR:                                                

         Cannot load Xdebug - it was already loaded

**Comment by sanmai** [comment]
And... There's a bug report #377 now.

File: appveyor.yml [hunk 1]
```diff
@@ -54,21 +54,12 @@ install:
...
-        $client = New-Object System.Net.WebClient
-        $xdebugUrl = "https://xdebug.org/files/php_xdebug-$env:xdebug_ver-nts-x86_64.dll"
-        $phpDir = (Get-Item (Get-Command php).Source).Directory.FullName
-        $xdebugPath = Join-Path $phpDir ext\xdebug.dll
-        $client.DownloadFile($xdebugUrl, $xdebugPath)
-        $phpIniPath = Join-Path $phpDir php.ini
-        Add-Content $phpIniPath @"
-        zend_extension=$xdebugPath
-        "@
+    - ps: if ($env:PHP -eq "1" ) { .ci\appveyor_install_xdebug.ps1 }
```
Comment (borNfreee) [inline appveyor.yml]: could you please explain what is it for, the main idea of this PR? I'm not sure I follow it

Comment (BackEndTea) [reply appveyor.yml]: We cache php, but before this change we applied xdebug on every single run, meaning we would have duplicate php ini entries for the xdebug configuration, which crashes phpspec.

The reason i moved it to a new file was due to strange windows errors with the`@"<text>"@` notation

Comment (sanmai) [reply appveyor.yml]: Could it be like this?
```diff
-        Add-Content $phpIniPath @"
-        zend_extension=$xdebugPath
-        "@
+       IF %PHP%==1 echo extension=php_openssl.dll >> $phpIniPath
```

Comment (BackEndTea) [reply appveyor.yml]: Well, if we don't have php cached we need to download xdebug, which is what this step also does.

Comment (sanmai) [reply appveyor.yml]: Then I guess that's the way to go.

**Comment by borNfreee** [comment]
thank you @BackEndTea

**Comment by BackEndTea** [comment]
Thank you @borNfreee  & @sanmai

**Pull request merged**